### ISSUE TITLE
SA1611 - Reported for documented parameter that is `ref struct`

### ISSUE THREAD

**Issue by smaillet** [DotNetAnalyzers/StyleCopAnalyzers#3933]
A simplified case of actual use that triggers SA1161 but should not

``` C#
/// <summary>Some docs</summary>
public ref struct MyRefStruct
{
    
    /// <summary>Initializes a new instance of the <see cref="MyRefStruct"/> struct.</summary>
    /// <param name="someParam">param docs...</param>
    public MyRefStruct(int someParam)
    {
        // ...
    }

    /// <summary>Dispose this instance</summary>
    [SuppressMessage( "Style", "IDE0251:Make member 'readonly'", Justification = "It mutates native state!" )]
    public void Dispose( )
    {
        // implement Dispose...
    }
}

public sealed class MyConsumer
{
    /// <summary>Performs some useful operation...</summary>
    /// <param name="foo">SA1161 triggers on this despite docs...</param>
    public void ConsumingMethod(ref readonly MyRefStruct foo)
    {
        // ...
    }
}
```

As listed in comments SA1161 triggers for `MyConsumer.ConsumingMethod` parameter `foo` even though it is documented.

**Comment by bjornhellander** [comment]
Can you provide a small but complete project which reproduces this behavior? It does not trigger for me.

**Comment by smaillet** [comment]
Sadly, small or public aren't an option at this point. I have observed that some cases it seems to be fine, but in the case that trigger this report it still reports an issue. I haven't been able to reduce it to isolate the source yet. (Nor is this code public, it is still very much "in development" and not worth being public yet.)
